servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/execution/GraphQLRequestHandler.kt

```diff
@@ -117,12 +115,13 @@ open class GraphQLRequestHandler(
         context: GraphQLContext?,
         graphQLContext: Map<*, Any>
     ): GraphQLBatchResponse {
-        val batchContext = getBatchContext(request, dataLoaderRegistryFactory?.generate())
+        val dataLoaderRegistry = dataLoaderRegistryFactory?.generate()
+        val batchContext = getBatchContext(request, dataLoaderRegistry)
         val batchGraphQLContext = graphQLContext + batchContext
         val responses = supervisorScope {
             request.requests.map {
                 async {
-                    execute(it, context, batchGraphQLContext)
+                    execute(it, context, batchGraphQLContext, dataLoaderRegistry)
```

Comment:
I don't think this is correct -> I believe that `batchContext` is empty then it should be NULL, otherwise yes its the one thats generated, see my previous comment